### ISSUE TITLE
[Feature Request]: Batch selection with actions

### ISSUE THREAD

**Issue by The-Underground-Club** [FreeTubeApp/FreeTube#2517]
### Guidelines

- [X] I have searched the [issue tracker](https://github.com/FreeTubeApp/FreeTube/issues) for a feature request that matches the one I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.

### Problem Description

Hi I would like to request a feature to 

- select all/select specific videos on subscriptions page with ability to:
    - batch copy links
    - batch download videos
    - batch mark as watched/unwatched

The reason I'm requesting this is because I've been watching videos on my iPhone on uYou, and I wanted to be able to mark all the videos I watched with ease, because right now it's a little bit of a pain. I also have been wanting to download certain videos in a playlist and going into each video and downloading is also very tedious.

### Proposed Solution

I would like the ability to select each individual video by Alt clicking, being able to batch select to a point by Shift clicking, or be able to select all videos by pressing "Cltr+A". Then when videos are selected, I would like to be able to batch copy video links, batch download videos, or batch mark video watched/unwatched. I would like this to be an option on the subscriptions page to bring the option to manage videos in your subscriptions.

### Alternatives Considered

Another solution would be able to add download option to the triple dots without entering video, mark all videos watched/unwatched in an entire specific channel.

### Issue Labels

ease of use improvement, improvement to existing feature, new feature, new keyboard shortcut

**Comment by jasonhenriquez** [comment]
Boosting this, as this would be a very useful feature for so many cases. Additional opinions:
- Select button on the top nav bar that changes hover/click behavior to "select" a video with a visible checkmark
- When "Select" mode is enabled, a "three dots" button appears on the top nav right next to it (disabled if no selection is made yet)
- Only applies to videos (if other types of results appear, e.g., playlists, hashtags, or channels in search results, they cannot be selected in Select mode)
- Each video's three-dot behavior only affects that video (no change)
- Copying multiple links: by default, has a newline delimiter between each link
- Clicking any "Favorite button" favorites all selected videos (note: this is somewhat imperfect; we really should add favoriting to the three-dot dropdown as well)

After playlist PR is merged, in separate PR from above:
- Applies to playlists as well
  - Only one type can be selected at a time; if one video is selected, only videos can be selected, and if one playlist is selected, only playlists can be selected
  - The top nav "three dots" button options populate with the appropriate options for the given type

**Comment by cdnia** [comment]
This will be very helpful, let's hope one day we get it.

**Comment by M0ahgra** [comment]
yes please do that it would make managing playlists so much faster
